pkg/cortex/modules.go

```diff
@@ -190,55 +190,109 @@ func (t *Cortex) initDistributor() (serv services.Service, err error) {
-
-	// Prometheus histograms for requests to the querier.
-	querierRequestDuration := promauto.With(prometheus.DefaultRegisterer).NewHistogramVec(prometheus.HistogramOpts{
-		Namespace: "cortex",
-		Name:      "querier_request_duration_seconds",
-		Help:      "Time (in seconds) spent serving HTTP requests to the querier.",
-		Buckets:   instrument.DefBuckets,
-	}, []string{"method", "route", "status_code", "ws"})
-
-	receivedMessageSize := promauto.With(prometheus.DefaultRegisterer).NewHistogramVec(prometheus.HistogramOpts{
-		Namespace: "cortex",
-		Name:      "querier_request_message_bytes",
-		Help:      "Size (in bytes) of messages received in the request to the querier.",
-		Buckets:   middleware.BodySizeBuckets,
-	}, []string{"method", "route"})
-
-	sentMessageSize := promauto.With(prometheus.DefaultRegisterer).NewHistogramVec(prometheus.HistogramOpts{
-		Namespace: "cortex",
-		Name:      "querier_response_message_bytes",
-		Help:      "Size (in bytes) of messages sent in response by the querier.",
-		Buckets:   middleware.BodySizeBuckets,
-	}, []string{"method", "route"})
-
-	inflightRequests := promauto.With(prometheus.DefaultRegisterer).NewGaugeVec(prometheus.GaugeOpts{
-		Namespace: "cortex",
-		Name:      "querier_inflight_requests",
-		Help:      "Current number of inflight requests to the querier.",
-	}, []string{"method", "route"})
-
-	// if we are not configured for single binary mode then the querier needs to register its paths externally
-	registerExternally := !t.Cfg.isModuleEnabled(All)
-	handler := t.API.RegisterQuerier(queryable, engine, t.Distributor, registerExternally, t.TombstonesLoader, querierRequestDuration, receivedMessageSize, sentMessageSize, inflightRequests)
-
-	// single binary mode requires a properly configured worker.  if the operator did not attempt to configure the
-	//  worker we will attempt an automatic configuration here
-	if t.Cfg.Worker.Address == "" && t.Cfg.isModuleEnabled(All) {
-		address := fmt.Sprintf("127.0.0.1:%d", t.Cfg.Server.GRPCListenPort)
-		level.Warn(util.Logger).Log("msg", "Worker address is empty in single binary mode.  Attempting automatic worker configuration.  If queries are unresponsive consider configuring the worker explicitly.", "address", address)
-		t.Cfg.Worker.Address = address
+
+	// Create a querier queryable and PromQL engine
+	t.QuerierQueryable, t.QuerierEngine = querier.New(t.Cfg.Querier, t.Overrides, t.Distributor, t.StoreQueryables, t.TombstonesLoader, querierRegisterer)
+
+	// Register the default endpoints that are always enabled for the querier module
+	t.API.RegisterQueryable(t.QuerierQueryable, t.Distributor)
+
+	return nil, nil
+}
+
+// initQuerier registers an internal HTTP router with a Prometheus API backed by the
+// Cortex Queryable. Then it does one of the following:
+// 1. Query-Frontend Enabled: If Cortex has an All or QueryFrontend target, the internal
```

Comment:
To make sure comment is formatted properly, please put empty lines between paragraphs (or list items), and don't use extra space at the beginning of the lines (except for diagram below) -- such lines are formatted as code according to Go doc conventions.

https://blog.golang.org/godoc

Reply:
I will update this to match the conventions

Reply:
Thanks. It helps tools to format it properly. (eg. GoLand :))